build_tools/cmake/iree_static_linker_test.cmake

```
@@ -0,0 +1,259 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
+
+# iree_static_linker_test()
+#
+# Creates a test with statically linked libraries. Assuming the target backend
+# is llvm-cpu and the driver is the local-sync driver.
+#
+# Parameters:
+#   NAME: Name of the target
+#   SRC: mlir source file to be compiled to an IREE module.
+#   INPUT_TYPE: Module input type, assuming native c/c++ data types
+#   HAL_TYPE: HAL data type, see runtime/src/iree/hal/buffer_view.h for possible
+#       options.
+#   STATIC_LIB_PREFIX: llvm static library prefix.
+#   MAIN_FUNCTION: Entry function call.
+#   EMITC: Optional, use emitC instead of vmfb module to define the vm contrl.
+#   COMPILER_FLAGS: additional flags to pass to the compiler. Bytecode output
+#       format and backend flags are passed automatically.
+#   INPUT_SHAPES: Module input shape, assuming "n, h, w, c" format. Support
+#       multiple entries.
+#   LABELS: Additional labels to apply to the test. The package path and
+#       "driver=local-sync" are added automatically.
+#   TARGET_CPU_FEATURES: If specified, a string passed as argument to
+#       --iree-llvm-target-cpu-features.
+#
+# Example:
+#   iree_static_linker_test(
+#     NAME
+#       edge_detection_test
+#     SRC
+#       "edge_detection.mlir"
+#     STATIC_LIB_PREFIX
+#       edge_detection_linked_llvm_cpu
+#     MAIN_FUNCTION
+#       "module.edge_detect_sobel_operator"
+#     INPUT_SHAPES
+#       "1, 128, 128, 1"
+#     INPUT_TYPE
+#       float
+#     HAL_TYPE
+#       IREE_HAL_ELEMENT_TYPE_FLOAT_32
+#     COMPILER_FLAGS
+#       "--iree-input-type=mhlo"
+#   )
+function(iree_static_linker_test)
+  if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  # See comment in iree_check_test about this condition.
+  if(NOT IREE_BUILD_COMPILER AND NOT CMAKE_CROSSCOMPILING)
+    return()
+  endif()
+
+  if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR NOT IREE_HAL_DRIVER_LOCAL_SYNC)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE
+    "EMITC"
+    "NAME;SRC;DRIVER;STATIC_LIB_PREFIX;MAIN_FUNCTION;INPUT_TYPE;HAL_TYPE"
+    "COMPILER_FLAGS;LABELS;TARGET_CPU_FEATURES;INPUT_SHAPES"
+    ${ARGN}
+  )
+
+  iree_get_executable_path(_COMPILER_TOOL "iree-compile")
+  get_filename_component(_SRC_PATH "${_RULE_SRC}" REALPATH)
+  iree_package_name(_PACKAGE_NAME)
+  iree_package_ns(_PACKAGE_NS)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  # Set up static library name.
+  set(_O_FILE_NAME "${_RULE_NAME}.o")
+  set(_H_FILE_NAME "${_RULE_NAME}.h")
+
+  # Set common iree-compile flags
+  set(_COMPILER_ARGS ${_RULE_COMPILER_FLAGS})
+  list(APPEND _COMPILER_ARGS "--iree-hal-target-backends=llvm-cpu")
+  list(APPEND _COMPILER_ARGS  "--iree-llvm-link-embedded=false")
+  list(APPEND _COMPILER_ARGS  "--iree-llvm-link-static")
```

nit: consistent spacing
```suggestion
  list(APPEND _COMPILER_ARGS "--iree-llvm-link-embedded=false")
  list(APPEND _COMPILER_ARGS "--iree-llvm-link-static")
```

---

Fixed.